internal/ringbuffer.go

```diff
@@ -0,0 +1,119 @@
+package internal
```

Comment:
Can you add the licence here.
You can add licence to default file template in your IDE.